dask/bag/text.py

```diff
@@ -61,11 +64,17 @@ def read_text(urlpath, blocksize=None, compression='infer',
     --------
     from_sequence: Build bag from Python sequence
     """
+    files = open_files(urlpath, mode='rt', encoding=encoding,
+                       errors=errors, compression=compression,
+                       **(storage_options or {}))
     if blocksize is None:
-        files = open_files(urlpath, mode='rt', encoding=encoding,
-                           errors=errors, compression=compression,
-                           **(storage_options or {}))
-        blocks = [delayed(list)(delayed(file_to_blocks)(fil)) for fil in files]
+        if files_per_partition is None:
+            blocks = [delayed(list)(delayed(file_to_blocks)(fil)) for fil in files]
```

Comment:
What happens if both `blocksize` and `files_per_partition` is set?

Reply:
I was thinking that in that case `files_per_partition` should just be ignored, as the point of `blocksize` is to have multiple partitions per file, so setting both is sort of invalid. Better to make it an explicit error, though.

Reply:
Might it make more sense to implement both in terms of `blocksize`? We generally know the size of the `OpenFile`s at this point, right? So we could use that to create partitions of the desired size in bytes, which would be better than the heuristic of nimbwr of files.

Reply:
I support unifying those to keywords if possible.  I'm curious if we do actually know the size of all of the files.  My guess is that this would require queries for each of them, which might be expensive if the data is remote.  cc @martindurant

Reply:
Indeed, getting the size of a file will generally need extra calls. In the case that blocksize is none, the only call necessary would have been to list the files, or if a list was provided up-front, then nothing at all.

Reply:
I'm not sure I understand the second statement.  I think the active question is "is it reasonably efficient to get the size of all files in `dask.bag.read_text`?"

Reply:
I would say that, in general, no.
If there is no compression (remember, we are talking about logical size here), and the url is a glob (so there will be a list operation, which also should get and cache sizes) then yes, depending on the specifics of the backend implementation. 
Since, when providing a set of specific files, the list operation isn't required, getting the size of each may bypass listing - this was a specific performance issue elsewhere - and so be a bigger fraction of the overhead.

Reply:
My particular use case is globbing uncompressed files in S3, so I benefit either way.

Pros of re-using `blocksize`:
* Simpler interface
* No weird condition of `blocksize is not None and files_per_partition is not None`
* Consistent partition sizes, even when file sizes aren't uniform

Pros of a new `files_per_partition` parameter:
* Simpler implementation
* Works for all cases, not just glob + no compression + backend enumeration returns sizes
* Easier to reason about (# partitions = # files / `files_per_partition`)

Given that we probably don't want to incur the penalty of measuring file sizes if they're not already available, I think I prefer the new parameter. Otherwise the speedup becomes a subtle thing that magically happens sometimes and not other times, or else we do and request the sizes, in which case it's potentially a slow-down if you're not careful. I think that the behavior and performance of a new parameter is simpler to reason about.

Reply:
> I think I prefer the new parameter

Same